engine.py

```diff
@@ -440,6 +447,36 @@ def __on_project_opened(self, opened_project, is_new):
         # change current engine context:
         sgtk.platform.change_context(ctx)
 
+    def __on_project_saved(self, saved_project):
+        if not self.__metadata_mgr.get_metadata(saved_project):
+            # this is not an sgtk compliant project
+            return
+
+        workfiles_app = self.apps.get("tk-multi-workfiles2")
+        proj_mgr_app = self.apps.get("tk-mari-projectmanager")
+        if not workfiles_app or not proj_mgr_app:
+            self.log_error("Unable to find workfiles or projectmanager app. Not exporting msf file.")
+            return
+
+        fields = self.context.as_template_fields()
+
+        # use project name to obtain "name" field
+        project_name_template = proj_mgr_app.get_template("template_new_project_name")
+        if not project_name_template.validate(saved_project.name()):
+            # this is not an sgtk compliant project
+            return
+        fields.update(project_name_template.get_fields(saved_project.name()))
+
+        # use project metadata to get the "version" field
+        # assume 1 if no version is set yet
+        fields["version"] = self.__metadata_mgr.get_project_version(saved_project) or 1
+
```

Comment:
Do not set a default version. If we fail work template validation then we should error out.

Reply:
The thing is, since v3 changes are already in use, and I am only setting new project version as 1 starting this pull request, earlier projects may have other metadata but not version (which should ideally have been 1) That's why I mentioned this is for backwards compatibility (in Github comment, not code)

Reply:
Then I would suggest actually setting the metadata in the `__on_project_opened` callback assuming the project is SGTK-ified already and just missing its version metadata.

Reply:
That works too.